Dotnet may fail when trying to update tools just after it was updated from package manager

<!--
Thanks for taking the time to fill out this bug report! 
Please make sure to 
[search for existing issues](https://github.com/topgrade-rs/topgrade/issues) 
before filing a new one!

Questions labeled with `Optional` can be skipped.
-->

<!-- 
If you're here to report about a "No asset found" error, please make sure that 
an hour has been passed since the last release was made. 
-->

## Erroneous Behavior
<!-- 
What actually happened?
-->
Dotnet step tried to update a tool "-------------------------------------" which is the separator, just after updating it from package manager.
I don't have any tools installed so it simply should not have tried running.

## Expected Behavior
<!-- 
Describe the expected behavior
-->
Dotnet step doesn't run because I don't have any tools installed.

## Steps to reproduce
<!-- 
A minimal example to reproduce the issue
-->
Run topgrade when a dotnet update is available through the package manager.

## Possible Cause (Optional)
<!-- 
If you know the possible cause of the issue, please tell us.
-->
When installing or updating dotnet through the package manager, it warns the user about a command running in the background to finish setuping it after the package manager is done running.
My guess is that in this case, the background command wasn't done when topgrade arrived to the dotnet step, and because of this there might have been more lines logged before the list of tools than usual (2, which are skipped in topgrade).

## Problem persists without calling from topgrade
<!-- 
Execute the erroneous command directly to see if the problem persists
-->
- [ ] Yes
- [ ] No

## Did you run topgrade through `Remote Execution`

- [ ] Yes
- [x] No

If yes, does the issue still occur when you run topgrade directlly in your
remote host

- [ ] Yes
- [ ] No

## Configuration file (Optional)
<!-- 
Paste your configuration file inside the code block if you think this issue is 
related to configuration.
-->

```toml

```

## Additional Details
- Operation System/Version
  <!-- For example, Fedora Linux 38 -->
Pop!_OS 22.04

- Installation
  <!-- 
  How did you install topgrade: build from repo / crates.io (cargo install topgrade) 
  / package manager (which one) / other (describe)
  -->
cargo/crates.io

- Topgrade version (`topgrade -V`)
12.0.2

## Verbose Output (`topgrade -v`)
<!-- 
Paste the verbose output into the pre-tags
-->

<details>
<pre>

</pre>
</details>

Ah, I just installed dotnet-SDK on my host, and reproduce it:

```sh
$ topgrade --only dotnet

── 09:03:23 - Sudo ─────────────────────────────────────────────────────────────

── 09:03:23 - .NET ─────────────────────────────────────────────────────────────
/tmp/4ffdd387-40aa-4171-a469-56e9e636ccb0/restore.csproj : error NU1101: Unable to find package ---------------------. No packages exist with this id in source(s): nuget.org
Tool '---------------------' failed to update due to the following:
The tool package could not be restored.
Tool '---------------------' failed to install. This failure may have been caused by:

* You are attempting to install a preview release and did not use the --version option to specify the version.
* A package by this name was found, but it was not a .NET tool.
* The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
* You mistyped the name of the tool.

For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool
.NET failed:
   0: Failed to update .NET package ---------------------
   1: Command failed: `/usr/bin/dotnet tool update --------------------- --global`
   2: `/usr/bin/dotnet` failed: exit status: 1

Location:
   src/steps/generic.rs:716

```

--- a/src/steps/generic.rs
+++ b/src/steps/generic.rs
@@ -676,6 +676,10 @@ pub fn run_dotnet_upgrade(ctx: &ExecutionContext) -> Result<()> {
         .run_type()
         .execute(&dotnet)
         .args(["tool", "list", "--global"])
+        // dotnet will print a greeting message on its first run, from this question:
+        // https://stackoverflow.com/q/70493706/14092446
+        // Setting `DOTNET_NOLOGO` to `true` should disable it
+        .env("DOTNET_NOLOGO", "true")
         .output_checked_utf8()
     {
         Ok(output) => output,

